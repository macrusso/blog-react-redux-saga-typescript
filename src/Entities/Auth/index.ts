import * as authConstants from './authConstants';
import * as authActions from './authActions';
import * as authSagas from './authSagas';
import authReducer, { ICommentsState } from './authReducer';

export { authSagas, ICommentsState, authActions, authReducer, authConstants };
