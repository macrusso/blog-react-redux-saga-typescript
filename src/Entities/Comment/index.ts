import * as commentConstants from './commentConstants';
import * as commentSelectors from './commentSelectors';
import * as commentActions from './commentActions';
import * as commentSagas from './commentSagas';
import commentReducer, { ICommentsState } from './commentReducer';
import ICommentPartial from './ICommentPartial';
import IComment from './IComment';

export {
  IComment,
  commentSagas,
  ICommentsState,
  commentActions,
  commentReducer,
  ICommentPartial,
  commentSelectors,
  commentConstants,
};
