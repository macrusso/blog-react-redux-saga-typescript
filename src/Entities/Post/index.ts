import * as postConstants from "./postConstants";
import * as postSelectors from "./postSelectors";
import * as postActions from "./postActions";
import * as postSagas from "./postSagas";
import postReducer from "./postReducer";
import Post from "../../Frontend/Post/PostListItem";
import IPostPartial from "./IPostPartial";
import IPost from "./IPost";
import IPostsState from "./IPostsState";

export { Post, IPost, postSagas, postActions, IPostsState, postReducer, IPostPartial, postSelectors, postConstants };
