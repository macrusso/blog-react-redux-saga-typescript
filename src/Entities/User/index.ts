import * as userConstants from './userConstants';
import * as userSelectors from './userSelectors';
import * as userActions from './userActions';
import * as userSagas from './userSagas';
import userReducer, { IUsersState } from './userReducer';
import IUserAuth from './IUserAuth';
import IUserPartial from './IUserPartial';
import IUser from './IUser';

export {
  IUser,
  IUserAuth,
  userSagas,
  userActions,
  IUsersState,
  userReducer,
  IUserPartial,
  userSelectors,
  userConstants,
};
