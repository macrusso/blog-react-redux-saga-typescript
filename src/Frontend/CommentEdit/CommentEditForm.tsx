import React from "react";
import { Formik } from "formik";
import { IComment } from "../../Entities";
import { TextField, Button, DialogActions, DialogContent } from "@material-ui/core";

interface ICommentEditFormProps {
  comment: IComment;
  handleCloseDialog: () => void;
  updateComment: (comment: IComment) => void;
}

const CommentEditForm: React.SFC<ICommentEditFormProps> = props => {
  const { comment, handleCloseDialog, updateComment } = props;
  return (
    <Formik
      initialValues={comment}
      onSubmit={values => {
        updateComment(values);
        handleCloseDialog();
      }}
    >
      {({ values, handleChange, handleBlur, handleSubmit }) => (
        <form onSubmit={handleSubmit} autoComplete="off">
          <DialogContent>
            <TextField
              id="body"
              label="Comment"
              name="body"
              fullWidth={true}
              margin="normal"
              variant="outlined"
              onChange={handleChange}
              value={values.body}
              onBlur={handleBlur}
              InputLabelProps={{
                shrink: true,
              }}
            />
          </DialogContent>
          <DialogActions>
            <Button onClick={handleCloseDialog} color="primary" variant="outlined">
              Cancel
            </Button>
            <Button type="submit" color="primary" variant="contained">
              Submit
            </Button>
          </DialogActions>
        </form>
      )}
    </Formik>
  );
};

export default CommentEditForm;
