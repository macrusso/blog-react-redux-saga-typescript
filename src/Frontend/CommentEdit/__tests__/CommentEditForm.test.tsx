import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import CommentEditForm from "../CommentEditForm";
import { commentTypes } from "../../../Entities";

describe("CommentEditForm", () => {
  test("Snapshot matches", () => {
    const comment: commentTypes.IComment= {
      _id: "5bcc554259e0dbfda6ed1a55",
      userId: "5bcc554259e0dbfda6ed1a55",
      postId: "5bcc554259e0dbfda6ed1a55",
      body: "string",
    };

    const wrapper = shallow(
      <BrowserRouter>
        <CommentEditForm comment={comment} updateComment={jest.fn()} handleCloseDialog={jest.fn()} />
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
