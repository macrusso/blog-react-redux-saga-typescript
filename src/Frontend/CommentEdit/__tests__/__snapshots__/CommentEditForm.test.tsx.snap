// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentEditForm Snapshot matches 1`] = `ShallowWrapper {}`;
